core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -761,6 +762,22 @@
     assertThat(s, isLinux(expected));
   }
 
+  @Test public void testExchange() {
+    final FrameworkConfig config = RelBuilderTest.config().build();
+    final RelBuilder builder = RelBuilder.create(config);
+    final RelNode rel = builder
+        .scan("EMP")
+        .exchange(RelDistributions.hash(ImmutableList.of(0, 1)))
+        .build();
+    String relJson = RelOptUtil.dumpPlan("", rel,
```

Comment:
final ?

Reply:
Well, actually, I'm not sure what's the benefit of using so much final in these tests.
But it seems it has no harm, so yes, it's better to add a final here.